model_analyzer/config/config.py

```diff
@@ -208,8 +210,8 @@ def _fill_config(self):
         self._add_config(
             ConfigField(
                 'objectives',
-                field_type=ConfigListString(),
-                default_value=['perf_throughput'],
+                field_type=objectives_scheme,
+                default_value={'perf_throughput': 10},
```

Comment:
There should be some default value for all of the metrics we use in the comparison. I am assuming that this is just for an initial starting point. You will be adding the other metrics later.

Reply:
We can change the default right now. What this means is that by default, the model analyzer will just order the results by throughput, highest to lowest. What do you suggest regarding the default? I can imagine adding a weightage for latency, but its relative value will be arbitrary.

Reply:
Why are you ordering anything? Nothing needs to be ordered. The integer weight values are normalized and can be added in any order.  As to what the defaults should be, that is something we need to experiment with. Remember, the weights are there to determine emphasis priority. If we think that latency is more important that GPU memory usage, then it should have a larger value. If we think it is 2x more important, then it should have a value 2x as large.

Reply:
Think of it this way. If a user does not specify custom weights, what values would MA use to determine the best run. Every metric that we use in the comparison heuristic will need a default weight value.

Reply:
Let me clarify, the above default value means that by default, the model analyzer will only compare the throughput values for each run, and sort the runs using that. This is because not only does the model analyzer give us the best result as defined by this weighting dictionary, it will give us the results in descending order of 'goodness'.  This way we have the best, the second best etc. 

If we haven't specified the weights for a metric, it is assumed to be 0. (That metric is not considered for comparison). We can experiment with and determine what an effective default should be.

Reply:
That might not be the best. The best throughput could be disastereous in every other metric.  But setting that aside, where are you setting the defaults to 0 for all other metrics?

Reply:
[Here](https://github.com/triton-inference-server/model_analyzer/blob/fe4bbe28d502413fc393db812bf87ced65f1d557/model_analyzer/result/result_comparator.py#L142) in the result comparator, there's a loop over the objectives specified in the dictionary. If the tag isn't present in this dictionary, the value isn't considered for comparison. This is equivalent to specifying it but setting its weight to 0.